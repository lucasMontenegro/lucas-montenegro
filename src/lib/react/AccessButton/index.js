import { makeStyles } from "@material-ui/core/styles"
import useAccessButtonState from "./useAccessButtonState"
import useTranslation from "lib/react/useTranslation"
import useResponsiveLayout from "lib/react/useResponsiveLayout"
import React, { Fragment } from "react"
import Button from "@material-ui/core/Button"
import Dialog from "@material-ui/core/Dialog"
import DialogContent from "@material-ui/core/DialogContent"
import DialogContentText from "@material-ui/core/DialogContentText"
import DialogActions from "@material-ui/core/DialogActions"
import FormControlLabel from "@material-ui/core/FormControlLabel"
import Typography from "@material-ui/core/Typography"
import Checkbox from "@material-ui/core/Checkbox"
import PropTypes from "prop-types"
const useStyles = makeStyles(theme => ({
  autoLogout: {
    paddingLeft: theme.spacing(2),
  },
  flexibleSpace: {
    flexGrow: 1,
  },
}), { name: `lib-react-access_button` })
const dialogId = `lib-react-access_button-logout_dialog`
const dialogTextId = `lib-react-access_button-logout_dialog_text`
export default function AccessButton (props) {
  const classes = useStyles()
  const state = useAccessButtonState(props.closeDashboard)
  const t = useTranslation()
  const r = useResponsiveLayout()
  return (
    <Fragment>
      {state.loggedIn ? (
        <Button
          color="default"
          variant="text"
          onClick={state.logout.open}
          aria-controls={dialogId}
        >
          {t({ en: () => `Log out`, es: () => `Salir` })}
        </Button>
      ) : (
        <Button color="primary" variant="contained" onClick={state.login}>
          {t({ en: () => `Log in`, es: () => `Ingresar` })}
        </Button>
      )}
      <Dialog
        maxWidth="sm"
        scroll="body"
        fullScreen={r({
          desktop: () => false,
          tablet: () => false,
          mobile: () => true,
        })}
        id={dialogId}
        aria-describedby={dialogTextId}
        open={state.logout.isOpen}
        onClose={state.logout.close}
      >
        <DialogContent>
          <DialogContentText color="textPrimary" variant="body1" id={dialogTextId}>
            {t({
              en: () => `Logging out may cause data loss.`,
              es: () => `Cerrar sesi√≥n puede causar perdida de datos.`,
            })}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <FormControlLabel
            className={classes.autoLogout}
            label={
              <Typography variant="body2" color="textSecondary" component="span">
                {t({
                  en: () => `Don't show me this again`,
                  es: () => `No volver a mostrarme esto`,
                })}
              </Typography>
            }
            control={
              <Checkbox
                color="primary"
                checked={state.logout.auto}
                onChange={state.logout.handleAuto}
              />
            }
          />
          <div className={classes.flexibleSpace} />
          <Button
            color="default"
            variant="text"
            onClick={state.logout.close}
            aria-controls={dialogId}
          >
            {t({ en: () => `Cancel`, es: () => `Cancelar` })}
          </Button>
          <Button autoFocus color="default" variant="text" onClick={state.logout.confirm}>
            {t({ en: () => `Ok`, es: () => `Aceptar` })}
          </Button>
        </DialogActions>
      </Dialog>
    </Fragment>
  )
}
AccessButton.propTypes = {
  closeDashboard: PropTypes.func,
}