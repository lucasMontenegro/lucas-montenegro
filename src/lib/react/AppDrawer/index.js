import React, { createContext, useContext, useState, useCallback } from "react"
import Drawer from "./Drawer"
import PropTypes from "prop-types"
export const AppDrawerContext = createContext(() => {})
AppDrawerContext.displayName = `AppDrawerContext`
export function useAppDrawerOpener () {
  return useContext(AppDrawerContext)
}
export default function AppDrawer (props) {
  const [isOpen, setIsOpen] = useState(false)
  return (
    <Drawer isOpen={isOpen} onClose={() => setIsOpen(false)} windowTitle={props.windowTitle}>
      <AppDrawerContext.Provider value={useCallback(() => setIsOpen(true), [setIsOpen])}>
        {props.children}
      </AppDrawerContext.Provider>
    </Drawer>
  )
}
AppDrawer.propTypes = { children: PropTypes.node }