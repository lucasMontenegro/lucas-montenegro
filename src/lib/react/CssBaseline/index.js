import "typeface-roboto"
export { default } from "@material-ui/core/CssBaseline"