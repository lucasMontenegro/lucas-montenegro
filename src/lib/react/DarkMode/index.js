import globals from "lib/utils/globals"
import React, { createContext, useContext, useState, useMemo } from "react"
const storageKey = `lib-react-dark_mode`
const initialValue = globals.window.localStorage.getItem(storageKey) === `dark`
export const DarkModeContext = createContext({
  value: initialValue,
  toggle () {},
})
DarkModeContext.displayName = `DarkModeContext`
export function useDarkMode () {
  return useContext(DarkModeContext)
}
export default function DarkMode (props) {
  const [value, saveValue] = useState(initialValue)
  const darkMode = useMemo(() => ({
    value,
    string: value ? `dark` : `light`,
    toggle () {
      globals.window.localStorage.setItem(storageKey, value ? `light` : `dark`)
      globals.setTimeout(() => saveValue(!value), 180)
    },
  }), [value, saveValue])
  return <DarkModeContext.Provider value={darkMode}>{props.children}</DarkModeContext.Provider>
}