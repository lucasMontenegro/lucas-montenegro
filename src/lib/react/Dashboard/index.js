import React, { createContext, useContext, useState, useCallback } from "react"
import Drawer from "./Drawer"
import PropTypes from "prop-types"
export const DashboardContext = createContext(() => {})
DashboardContext.displayName = `DashboardContext`
export function useDashboardOpener () {
  return useContext(DashboardContext)
}
export default function Dashboard (props) {
  const [isOpen, setIsOpen] = useState(false)
  return (
    <Drawer isOpen={isOpen} onClose={() => setIsOpen(false)}>
      <DashboardContext.Provider value={useCallback(() => setIsOpen(true), [setIsOpen])}>
        {props.children}
      </DashboardContext.Provider>
    </Drawer>
  )
}
Dashboard.propTypes = { children: PropTypes.node }