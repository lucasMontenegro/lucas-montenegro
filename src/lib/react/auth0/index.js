import React, { createContext, useContext, useState, useEffect, useMemo, Fragment } from "react"
import initAuth0 from "./initAuth0"
import useLoginTimeoutAlert from "./useLoginTimeoutAlert"
import makeLoginFunction from "./makeLoginFunction"
import makeLogoutFunction from "./makeLogoutFunction"
import PropTypes from "prop-types"
export const Auth0Context = createContext()
Auth0Context.displayName = `Auth0Context`
export function useAuth0 () {
  return useContext(Auth0Context)
}
export function Auth0Provider (props) {
  const [client, setClient] = useState(null)
  const [user, setUser] = useState(null)
  useEffect(() => {
    initAuth0(setClient, setUser)
  }, [])
  const lta = useLoginTimeoutAlert()
  const openLta = lta.open
  const { getLogoutUrl } = props
  return (
    <Auth0Context.Provider
      value={useMemo(() => ({
        user,
        login: makeLoginFunction(client, openLta, setUser),
        logout: makeLogoutFunction(user, client, getLogoutUrl),
      }), [user, client, openLta, getLogoutUrl])}
    >
      {client === null ? null : (
        <Fragment>
          {lta.node}
          {props.children}
        </Fragment>
      )}
    </Auth0Context.Provider>
  )
}
Auth0Provider.propTypes = {
  getLogoutUrl: PropTypes.func.isRequired,
  children: PropTypes.node,
}