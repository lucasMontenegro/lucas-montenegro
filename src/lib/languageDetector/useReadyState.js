import { useLayoutEffect, useState } from "react"
import i18next from "i18next"
export default function useReadyState () {
  const [ready, setReady] = useState(false)
  useLayoutEffect(() => {
    if (i18next.isInitialized) {
      setReady(true)
    } else {
      function handleInit () {
        setReady(true)
      }
      i18next.on(`initialized`, handleInit)
      return () => i18next.off(`initialized`, handleInit)
    }
  }, [])
  return ready
}