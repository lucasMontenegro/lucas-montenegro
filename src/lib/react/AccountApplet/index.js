import { makeStyles } from "@material-ui/core/styles"
import { useAuth0 } from "lib/react/auth0"
import useTranslation from "lib/react/useTranslation"
import React from "react"
import Button from "@material-ui/core/Button"
import ProfileLinks from "./ProfileLinks"
import Layout from "./Layout"
import DefaultAvatarSvg from "./DefaultAvatarSvg"
import Link from "lib/react/links/Link"
import PropTypes from "prop-types"
const useStyles = makeStyles(theme => ({
  logoutButton: {
    fontSize: theme.spacing(1.5),
    lineHeight: 1.5,
    backgroundColor: (
      theme.palette.type === `dark` ? `rgba(255, 255, 255, 0.12)` :
      `rgba(0, 0, 0, 0.1)`
    ),
    "&:hover": {
      backgroundColor: (
        theme.palette.type === `dark` ? `rgba(255, 255, 255, 0.16)` :
        `rgba(0, 0, 0, 0.14)`
      ),
    },
  },
  createProfile: {
    display: `inline-block`,
    border: `${theme.spacing(0.25)}px solid ${theme.palette.primary.main}`,
    borderRadius: theme.spacing(0.5),
    padding: `0 ${theme.spacing(1.5)}px`,
  },
}), { name: `lib-react-account_applet` })
export default function AccountApplet (props) {
  const classes = useStyles()
  const auth0 = useAuth0()
  const t = useTranslation()
  const { user } = auth0
  const { closeDashboard } = props
  if (user) {
    const logoutButton = (
      <Button
        className={classes.logoutButton}
        size="small"
        variant="text"
        color="default"
        onClick={auth0.logout}
      >
        {t({ en: () => `Log out`, es: () => `Salir` })}
      </Button>
    )
    const { profile_id } = user
    if (profile_id) {
      return (
        <ProfileLinks
          t={t}
          closeDashboard={closeDashboard}
          logoutButton={logoutButton}
          profileId={profile_id}
        />
      )
    } else {
      return (
        <Layout
          avatar={<DefaultAvatarSvg />}
          topText={
            <Link to="#" className={classes.createProfile} onClick={closeDashboard}>
              {t({ en: () => `Create profile`, es: () => `Crear perfil` })}
            </Link>
          }
          bottomText={logoutButton}
        />
      )
    }
  }
  return (
    <Layout
      avatar={<DefaultAvatarSvg />}
      text={
        <Button
          variant="contained"
          color="primary"
          onClick={() => {
            auth0.login()
            closeDashboard()
          }}
        >
          {t({ en: () => `Log in`, es: () => `Ingresar` })}
        </Button>
      }
    />
  )
}
AccountApplet.propTypes = {
  closeDashboard: PropTypes.func,
}