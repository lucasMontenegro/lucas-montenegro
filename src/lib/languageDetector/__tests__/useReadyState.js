import { useLayoutEffect, useState } from "react"
import i18next from "i18next"
import useReadyState from "lib/languageDetector/useReadyState"
jest.mock(`react`, () => ({
  __esModule: true,
  useState: jest.fn(),
  useLayoutEffect: jest.fn(),
}))
let ready
const setReady = jest.fn()
useState.mockImplementation(() => [ready, setReady])
jest.mock(`i18next`, () => ({
  __esModule: true,
  default: { on: jest.fn(), off: jest.fn() },
}))
describe(`lib/languageDetector/useReadyState`, () => {
  it(`should use the right dependency versions`, () => {
    expect(jestUtils.getDependencies([`react`, `i18next`])).toMatchSnapshot()
  })
  {
    const msg = `useReadyState first render (ready false, isInitialized %j)`
    describe.each([[true], [false]])(msg, initialized => {
      let result, effect
      const getEffect = () => effect || (effect = useLayoutEffect.mock.calls[0][0])
      beforeAll(() => {
        ready = false
        i18next.isInitialized = initialized
        result = useReadyState()
      })
      it(`should set up the initial state`, () => {
        expect(useState.mock.calls).toEqual([[false]])
        useState.mockClear()
      })
      it(`should set up the layout effect`, () => {
        expect(useLayoutEffect.mock.calls).toHaveLength(1)
        expect(useLayoutEffect.mock.calls[0]).toHaveLength(2)
        expect(getEffect()).toBeInstanceOf(Function)
        expect(useLayoutEffect.mock.calls[0][1]).toEqual([])
        useLayoutEffect.mockClear()
      })
      it(`should not call any other function`, () => {
        expect(setReady.mock.calls).toEqual([])
        expect(i18next.on.mock.calls).toEqual([])
        expect(i18next.off.mock.calls).toEqual([])
      })
      describe(`layout effect (isInitialized true)`, () => {
        let clearEffect
        beforeAll(() => {
          i18next.isInitialized = true
          clearEffect = getEffect()()
        })
        it(`should update the state`, () => {
          expect(setReady.mock.calls).toEqual([[true]])
          setReady.mockClear()
        })
        it(`should not call any other function`, () => {
          expect(useState.mock.calls).toEqual([])
          expect(useLayoutEffect.mock.calls).toEqual([])
          expect(i18next.on.mock.calls).toEqual([])
          expect(i18next.off.mock.calls).toEqual([])
        })
        it(`should not return anything`, () => {
          expect(clearEffect).toBeUndefined()
        })
      })
      describe(`layout effect (isInitialized false)`, () => {
        let clearEffect, handleInit
        const getInitHandler = () => handleInit || (handleInit = i18next.on.mock.calls[0][1])
        beforeAll(() => {
          i18next.isInitialized = false
          clearEffect = getEffect()()
        })
        it(`should handle "on initialized" events`, () => {
          expect(i18next.on.mock.calls).toHaveLength(1)
          expect(i18next.on.mock.calls[0]).toHaveLength(2)
          expect(i18next.on.mock.calls[0][0]).toBe(`initialized`)
          expect(getInitHandler()).toBeInstanceOf(Function)
          i18next.on.mockClear()
        })
        it(`should not call any other function`, () => {
          expect(useState.mock.calls).toEqual([])
          expect(setReady.mock.calls).toEqual([])
          expect(useLayoutEffect.mock.calls).toEqual([])
          expect(i18next.off.mock.calls).toEqual([])
        })
        it(`should return an effect cleanup function`, () => {
          expect(clearEffect).toBeInstanceOf(Function)
        })
        describe(`"on initialized" event handler`, () => {
          // at this point i18next.isInitialized is always true
          beforeAll(() => getInitHandler()())
          it(`should update the state`, () => {
            expect(setReady.mock.calls).toEqual([[true]])
            setReady.mockClear()
          })
          it(`should not call any other function`, () => {
            expect(useState.mock.calls).toEqual([])
            expect(useLayoutEffect.mock.calls).toEqual([])
            expect(i18next.on.mock.calls).toEqual([])
            expect(i18next.off.mock.calls).toEqual([])
          })
        })
        {
          const msg = `effect cleanup handler (isInitialized %j)`
          describe.each([[true], [false]])(msg, initialized => {
            beforeAll(() => {
              i18next.isInitialized = initialized
              clearEffect()
            })
            it(`should set off the event handler`, () => {
              expect(i18next.off.mock.calls).toHaveLength(1)
              expect(i18next.off.mock.calls[0]).toHaveLength(2)
              expect(i18next.off.mock.calls[0][0]).toBe(`initialized`)
              expect(i18next.off.mock.calls[0][1]).toBe(getInitHandler())
              i18next.off.mockClear()
            })
            it(`should not call any other function`, () => {
              expect(useState.mock.calls).toEqual([])
              expect(setReady.mock.calls).toEqual([])
              expect(useLayoutEffect.mock.calls).toEqual([])
              expect(i18next.off.mock.calls).toEqual([])
            })
          })
        }
      })
    })
  }
  {
    const msg = `useReadyState re-render (ready %j, isInitialized %j)`
    const cases = [
      [true, true],
      [true, false],
      [false, true],
      [false, false],
    ]
    describe.each(cases)(msg, (readyValue, isInitialized) => {
      let result
      beforeAll(() => {
        i18next.isInitialized = isInitialized
        ready = readyValue
        result = useReadyState()
      })
      it(`should read from the state`, () => {
        expect(useState.mock.calls).toEqual([[false]])
        useState.mockClear()
      })
      it(`should set up the layout effect`, () => {
        expect(useLayoutEffect.mock.calls).toHaveLength(1)
        expect(useLayoutEffect.mock.calls[0]).toHaveLength(2)
        expect(useLayoutEffect.mock.calls[0][0]).toBeInstanceOf(Function)
        expect(useLayoutEffect.mock.calls[0][1]).toEqual([])
        useLayoutEffect.mockClear()
      })
      it(`should not call any other function`, () => {
        expect(setReady.mock.calls).toEqual([])
        expect(i18next.on.mock.calls).toEqual([])
        expect(i18next.off.mock.calls).toEqual([])
      })
      it(`should return the value of "ready"`, () => {
        expect(result).toBe(readyValue)
      })
    })
  }
})