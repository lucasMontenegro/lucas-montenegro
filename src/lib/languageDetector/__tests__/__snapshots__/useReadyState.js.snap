// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/languageDetector/useReadyState should use the right dependency versions 1`] = `
Object {
  "i18next": "^17.0.18",
  "react": "^16.10.1",
}
`;
